bin/reth/src/node/mod.rs

```diff
@@ -102,7 +106,14 @@ impl Command {
         let genesis_hash = init_genesis(db.clone(), self.chain.genesis.clone())?;
 
         info!("Connecting to p2p");
-        let network = start_network(network_config(db.clone(), chain_id, genesis_hash)).await?;
+        let network = start_network(
+                network_config(
+                    db.clone(), 
+                    chain_id, 
+                    genesis_hash,
+                    self.disable_discovery,
```

Comment:
I suspect we'll have a bunch more of these, so should switch to builder pattern, but fine for now.